RoaringBitmap/src/main/java/org/roaringbitmap/longlong/LongUtils.java

```diff
@@ -171,15 +150,10 @@ public static byte[] highPartInPlace(long num, byte[] high48) {
    * checks if given high48 is the maximum possible one
    * (e.g. it is the case for -1L, which is the maximum unsigned long)
    *
-   * @param high48 the byte array
+   * @param key long
    * @return true if this the maximum high part
    */
-  public static boolean isMaxHigh(byte[] high48) {
-    return high48[0] == -1
-            && high48[1] == -1
-            && high48[2] == -1
-            && high48[3] == -1
-            && high48[4] == -1
-            && high48[5] == -1;
+  public static boolean isMaxHigh(long key) {
```

Comment:
This does not seem to be equivalent to previous code at all! You are effectively checking that the key is exactly equal to 0xffff (and it does not match the comment above the function), whereas the previous code check that 40 bytes are set to 0xff. It is possible that it is equivalent, but we would need an analysis to prove that it is.

I submit to you that `~0 >>> 16` is a cumbersome way to write 0xffff. Please prefer 0xffff or 0xFFFF or even 65535.